File: src/main/java/um/haberes/rest/model/CargoClase.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@EqualsAndHashCode(callSuper=false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CargoClase extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 4113850296413347255L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cargoClaseId;

	private String nombre = "";
	private Integer claseId;
	
	@OneToOne
	@JoinColumn(name = "claseId", insertable = false, updatable = false)
	private Clase clase;

}

File: src/main/java/um/haberes/rest/kotlin/model/CargoClase.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*

@Entity
@Table
data class CargoClase(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cargoClaseId: Long? = null,

    var nombre: String = "",
    var claseId: Int? = null,

    @OneToOne
    @JoinColumn(name = "claseId", insertable = false, updatable = false)
    var clase: Clase? = null

) : Auditable()
